lib/grape_oas/api_model_builders/response.rb

```diff
@@ -62,14 +62,64 @@ def group_specs_by_status(specs)
 
       # Builds a response from a group of specs with the same status code
       # If multiple specs have `as:` keys, they are merged into a single object schema
+      # If any spec has `one_of:` key, they are merged into a oneOf schema
+      # If mixed oneOf and as: keys, falls back to merged object behavior
       def build_response_from_group(group_specs)
-        if group_specs.any? { |s| s[:as] }
+        has_one_of = group_specs.any? { |s| s[:one_of] }
+        has_as = group_specs.any? { |s| s[:as] }
+
+        if has_one_of && !has_as
+          build_one_of_response(group_specs)
+        elsif has_as
           build_merged_response(group_specs)
         else
           build_response_from_spec(group_specs.first)
         end
       end
 
+      # Builds a oneOf response for multiple possible response schemas
+      def build_one_of_response(specs)
+        first_spec = specs.first
+
+        all_schemas = []
+        specs.each do |spec|
+          if spec[:one_of]
+            spec[:one_of].each do |one_of_spec|
+              entity_schema = build_schema(one_of_spec[:model] || one_of_spec[:entity])
+
+              if one_of_spec[:is_array]
+                entity_schema = GrapeOAS::ApiModel::Schema.new(
+                  type: Constants::SchemaTypes::ARRAY,
+                  items: entity_schema,
+                )
+              end
+
+              all_schemas << entity_schema
+            end
+          else
+            # Handle regular specs mixed with oneOf specs
+            entity_schema = build_schema(spec[:entity])
+            all_schemas << entity_schema if entity_schema
+          end
```

Comment:
To count `is_array` properly, consider honoring it for both `one_of` entries and plain specs, e.g.:
```suggestion
          if spec[:one_of]
            spec[:one_of].each do |one_of_spec|
              is_array = one_of_spec.key?(:is_array) ? one_of_spec[:is_array] : spec[:is_array]
              schema = build_schema(one_of_spec[:model] || one_of_spec[:entity])
              schema = array_schema(schema) if is_array
              all_schemas << schema if schema
            end
          else
            schema = build_schema(spec[:entity])
            schema = array_schema(schema) if spec[:is_array]
            all_schemas << schema if schema
          end
```

You could add a small helper:
```ruby
def array_schema(schema)
  GrapeOAS::ApiModel::Schema.new(
    type: Constants::SchemaTypes::ARRAY,
    items: schema,
  )
end
```